src/storage/txn/commands.rs

```diff
@@ -278,23 +288,59 @@ macro_rules! command {
                 .into()
             }
         }
+    }
+}
 
-        impl CommandExt for $cmd {
-            fn tag(&self) -> metrics::CommandKind {
-                metrics::CommandKind::$tag
-            }
+macro_rules! ts {
+    ($ts:ident) => {
+        fn ts(&self) -> TimeStamp {
+            self.$ts
+        }
+    };
+}
 
-            fn incr_cmd_metric(&self) {
-                KV_COMMAND_COUNTER_VEC_STATIC.$tag.inc();
-            }
+macro_rules! tag {
+    ($tag:ident) => {
+        fn tag(&self) -> metrics::CommandKind {
+            metrics::CommandKind::$tag
+        }
 
-            $(
-            fn ts(&self) -> TimeStamp {
-                self.$ts
-            }
-            )?
+        fn incr_cmd_metric(&self) {
+            KV_COMMAND_COUNTER_VEC_STATIC.$tag.inc();
         }
-    }
+    };
+}
+
+macro_rules! readonly {
```

Comment:
The following 4 macros seems so similar that I think we can abstract them into one.
But I cannot find a good name for this single macro (maybe `field!` ?).

Reply:
`command_method` or `cmd_method`?